lib/recomb_map.c

```diff
@@ -196,6 +208,31 @@ recomb_map_phys_to_discrete_genetic(recomb_map_t *self, double x, uint32_t *locu
     return ret;
 }
 
+/* Find the `index` of the interval within `values` the `query` fits, such that
+ * values[index-1] < query <= values[index]
+ * If necessary, will find the leftmost such index
+ */
+size_t
+recomb_map_binary_interval_search(double query, double *values, size_t n_values)
```

Comment:
Since this isn't a "method" of recomb_map (as the first parameter isn't a ``recomb_map_t``) we should rename it.  It's just a binary search, isn't it, where we find the leftmost value <= query? So, ``msp_binary_search``, perhaps?